allennlp/common/registrable.py

```diff
@@ -39,14 +39,31 @@ class Registrable(FromParams):
     default_implementation: str = None
 
     @classmethod
-    def register(cls: Type[T], name: str):
+    def register(cls: Type[T], name: str, exist_ok=False):
+        """
+        Register a class under a particular name.
+
+        Parameters
+        ----------
+        name: ``str``
+            The name to register the class under.
+        exist_ok: ``bool`, optional (default=False)
+            If True, overwrites any existing models registered under ``name``. Else,
+            throws an error if a model is already registered under ``name``.
+        """
         registry = Registrable._registry[cls]
         def add_subclass_to_registry(subclass: Type[T]):
             # Add to registry, raise an error if key has already been used.
             if name in registry:
-                message = "Cannot register %s as %s; name already in use for %s" % (
-                        name, cls.__name__, registry[name].__name__)
-                raise ConfigurationError(message)
+                if exist_ok:
+                    message = ("%s has already been registered as %s, but "
+                               "exist_ok=True, so overwriting with %s" % (
+                                       name, registry[name].__name__, cls.__name__))
```

Comment:
can you change this to use f-strings (it would be more readable) (I know it was already like this)